s3/src/main/scala/akka/stream/alpakka/s3/scaladsl/S3Client.scala

```diff
@@ -41,6 +44,104 @@ final case class ListBucketResultContents(
     storageClass: String
 )
 
+/**
+ * Modelled after com.amazonaws.services.s3.model.ObjectMetadata
+ * @param metadata the raw http headers
+ */
+final case class ObjectMetadata(
```

Comment:
Maybe not make it a case class as that adds a bunch of things to the api, `copy` for example which I think does not do make sense for this. Alternatively make a trait describing the public API and use a case class to implement it.